test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin_test.cc

```diff
@@ -0,0 +1,146 @@
+  CommandLineOptions options;
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  MetricsPluginPtr plugin = config_factory.createMetricsPlugin(config_any);
+  EXPECT_NE(dynamic_cast<FakeMetricsPlugin*>(plugin.get()), nullptr);
+}
+
+TEST(FakeMetricsPluginConfigFactory, ValidateConfigWithBadConfigProtoReturnsError) {
+  Envoy::ProtobufWkt::Any empty_any;
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  absl::Status status = config_factory.ValidateConfig(empty_any);
+  EXPECT_THAT(status.message(), HasSubstr("Failed to parse"));
+}
+
+TEST(FakeMetricsPluginConfigFactory, ValidateConfigWithWellFormedIllegalConfigReturnsError) {
+  FakeMetricsPluginConfig config;
+  // Negative value fails config validation:
+  config.set_fixed_metric_value(-1.0);
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  absl::Status status = config_factory.ValidateConfig(config_any);
+  EXPECT_THAT(status.message(), HasSubstr("Negative fixed_metric_value"));
+}
+
+TEST(FakeMetricsPluginConfigFactory, ValidateConfigWithDefaultConfigReturnsOk) {
+  FakeMetricsPluginConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  absl::Status status = config_factory.ValidateConfig(config_any);
+  EXPECT_TRUE(status.ok());
+}
+
+TEST(FakeMetricsPluginConfigFactory, ValidateConfigWithValidConfigReturnsOk) {
+  FakeMetricsPluginConfig config;
+  config.set_fixed_metric_value(1.0);
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  absl::Status status = config_factory.ValidateConfig(config_any);
+  EXPECT_TRUE(status.ok());
+}
+
+TEST(FakeMetricsPlugin, GetMetricByNameReturnsDefaultValueForGoodMetricName) {
+  FakeMetricsPlugin metrics_plugin(FakeMetricsPluginConfig{});
+  absl::StatusOr<double> metric_value_or = metrics_plugin.GetMetricByName("good-metric");
+  ASSERT_TRUE(metric_value_or.ok());
+  EXPECT_EQ(metric_value_or.value(), 0.0);
+}
+
+TEST(FakeMetricsPlugin, GetMetricByNameReturnsValueFromConfigForGoodMetricName) {
+  FakeMetricsPluginConfig config;
+  config.set_fixed_metric_value(5678.0);
+  FakeMetricsPlugin metrics_plugin(config);
+  absl::StatusOr<double> metric_value_or = metrics_plugin.GetMetricByName("good-metric");
+  ASSERT_TRUE(metric_value_or.ok());
+  EXPECT_EQ(metric_value_or.value(), 5678.0);
```

Comment:
This magic number re-emerging here could even be shared  x-test! :-)

Reply:
Got rid of some of these magic numbers, replaced with 0.0.